src/azure-cli-core/azure/cli/core/__init__.py

```diff
@@ -499,7 +499,7 @@ def load_arguments(self, command=None):
             command_loaders = self.cmd_to_loader_map.get(command, None)
 
         if command_loaders:
-            for loader in command_loaders:
+            for loader in sorted(command_loaders, key=lambda loader: (0 if loader.__module__.startswith('azext') else 1, loader.__module__, loader.__class__.__name__)):
```

Comment:
I think there is a side effect if we prioritize extensions. Consider the reserved case described in https://github.com/Azure/azure-cli/pull/27296#issuecomment-1878581800:

> A command module supports `_subscription` but an extension ignores it. In that case, if the extension is loaded first, the help message for the command module will not have `_subscription` which is wrong.